Add a quiet time for beep hour

dijkstrw opened the pull request:
After I'm done it feels kinda big for what it is. Feel free to tell me how this can be improved.

dijkstrw committed:
Add a quiet time for beep hour

carrotIndustries commented on common/app/conf/app.c:
```diff
@@ -140,6 +140,92 @@ static const svc_menu_item_text_t menu_item_hourbeep_test = {
+	return hourbeep_quiet_get_time(1);
+}
+
+static void hourbeep_quiet_set_time(uint8_t start, int8_t dig, uint8_t dir) {
```
For consistency reasons I'd prefer using the CLAMP macro in this place as it is used other places in the firmware. ADD_MOD also leads to somewhat unintuitive operation: hour is 17, user increments 10's digit -> 3

carrotIndustries commented on common/app/conf/app.c:
```diff
@@ -140,6 +140,92 @@ static const svc_menu_item_text_t menu_item_hourbeep_test = {
+	svc_beep_hour_quiet_set_time(start, bq.h, bq.m);
+}
+
+static void hourbeep_quiet_set_start_time(uint8_t dig, int8_t dir, void *ud) {
```
You can merge this function with the corresponding stop function into one and set the user_data that gets passed in the respective svc_menu_item_adj_t.

carrotIndustries commented on common/svc/beep.c:
```diff
@@ -43,17 +44,38 @@ void svc_beep_key(void) {
 static uint16_t SECTION_INFOMEM beep_hour_duration = 10;
-
+static uint8_t  SECTION_INFOMEM beep_hour_quiet_enable = 0;
+static svc_beep_hour_quiet_t beep_hour_quiet[2] = {
```
This could go in the nonvolatile SECTION_INFOMEM as well.

carrotIndustries commented on common/svc/beep.c:
```diff
@@ -43,17 +44,38 @@ void svc_beep_key(void) {
 
 void svc_beep_hour(void) {
 	static uint8_t hour_last = 255;
-	
```
This looks a bit too complicated to me. Why not ditch `beep_hour_muted` and simply check whether the quiet hours are enabled / active before beeping?

carrotIndustries commented:
Looks good to me overall!
I don't mind the size of this commit at all, that's what you get when following the architecture.

dijkstrw committed:
Add a quiet interval for beep hour

Take #2;
- Use only hours to indicate begin/end, simplifies code and menu
- Ditch beep_hour_muted
- CLAMP instead of ADD_MOD

dijkstrw commented on common/svc/beep.c:
```diff
@@ -45,11 +45,7 @@ static uint16_t SECTION_INFOMEM beep_hour_freq = 2000;
-	{.h=06, .m=59}
-};
-static uint8_t beep_hour_muted = 0;
+static uint8_t  beep_hour_quiet[2] = {21, 06};
```
SECTION_INFOMEM seems full?

carrotIndustries replied:
Oh, I see. Just decrease the number of alarms to something more sensible like 10 or so.

carrotIndustries commented on common/svc/beep.c:
```diff
@@ -104,3 +122,20 @@ void svc_beep_hour_set_freq(uint16_t f) {
+
+uint16_t svc_beep_hour_quiet_get_interval(void) {
+	return ((beep_hour_quiet[0] * 100) + beep_hour_quiet[1]);
+}
```
I'm not exactly happy with this one: Turning otherwise unrelated numbers (like hours and minutes) into a "string" number (minutes+hours*100) is an oddity of `svc_menu_item_adj_t` and shouldn't spill into other layers of the firmware. My suggestion: `void svc_beep_hour_quiet_get_interval(uint8_t *s, uint8_t *e)`

dijkstrw committed:
Beep quiet interval clean up

- reduce number of alarms to 10 to free up SECTION_INFOMEM

- use SECTION_INFOMEM for permanent beep quiet interval storage

- remove menu item structure knowledge from beep service

carrotIndustries commented:
somehow, github didn't recognize the squashed merge...
